### DISCUSSION TITLE
Is there a way to return 404 page if using default prefix in url?

### DISCUSSION THREAD

**Discussion by websorokinweb** [vercel/next.js#59980] [App Router]
Desired behaviour: 
domain.com - handles pages with default locale (ua)
domain.com/ua - returns 404

Is that even possible without redirect?

My next config:
i18n: {
    locales: ['ua', 'en'],
    localeDetection: false,
    defaultLocale: 'ua',
  },

**Comment by Baryonae** [comment]
i guess this would help?
https://nextjs.org/docs/pages/building-your-application/routing/custom-error

**Reply by websorokinweb** [reply]
Thank you for pointing, but it is not what I am searching for. If I get it correctly, it just helps to customize 404 page ui. What I want to get is trigger that page 404 on every /ua/**

**Reply by Baryonae** [reply]
@websorokinweb are you using app router or src / ??

**Reply by websorokinweb** [reply]
I am using /pages router

**Reply by websorokinweb** [reply]
@Avi98 I don't get it. Whenever it's domain.com or domain.com/ua I just get domain.com/ua in nextUrl

**Reply by Baryonae** [reply]
well you should update to app routing, u got a advantage there like if you create a folder like app/ua and put a loading file there like app/ua/404.tsx ig it should show the page whenever you go to a unknown directory in the /ua 
more info in the doc I sent

**Reply by websorokinweb** [reply]
It isn't so easy unfortunately: when I type in address domain.com or domain.com/ua I get it middleware as domain.com/ua either. 

@Baryonae It is too much work to switch into app router unfortunately. But your solution might work
